test/Test/Toml/Gen.hs

```diff
@@ -150,9 +152,41 @@ genZoned = do
     pure $ ZonedTime local zTime
 
 genDate :: MonadGen m => m DateTime
-genDate = Gen.choice 
+genDate = Gen.choice
     [ Day   <$> genDay
     , Hours <$> genHours
     , Local <$> genLocal
     , Zoned <$> genZoned
-    ]
\ No newline at end of file
+    ]
+
+genBool :: MonadGen m => m Bool
+genBool = Gen.bool
+
+genInteger :: MonadGen m => m Integer
+genInteger = toInteger <$> Gen.int (Range.constantBounded @Int)
+
+genDouble :: MonadGen m => m Double
+genDouble = Gen.double $ Range.constant @Double (-1000000.0) 1000000.0
+
+genText :: MonadGen m => m Text
+genText = Gen.text (Range.constant 0 256) Gen.alphaNum
+
+noneArrayList :: MonadGen m => [m AnyValue]
+noneArrayList =
+    [ AnyValue . Bool    <$> genBool
+    , AnyValue . Integer <$> genInteger
+    , AnyValue . Double  <$> genDouble
+    , AnyValue . Text    <$> genText
+    , AnyValue . Date    <$> genDate
+    ]
+
+genArrayFrom :: MonadGen m => m AnyValue -> m (Value 'TArray)
+genArrayFrom noneArray = fromMaybe (error "Error in genArrayFrom") . toMArray <$> Gen.list (Range.constant 3 6) noneArray
+
+genArray :: MonadGen m => m (Value 'TArray)
+genArray = Gen.recursive Gen.choice
+    [Gen.choice $ map genArrayFrom noneArrayList]
```

Comment:
I think that this argument of `recursive` function should be `map genArrayFrom noneArrayList` instead of `[Gen.choice $ map genArrayFrom noneArrayList]`. This argument contains list of non-recursive generators. In case of array generation, it should be either array of integers or array of doubles or array of text, etc. Don't need to apply `choice` additionally.

Reply:
It does`t work without `Gen.choice` that giving `[m a]`, otherwise incorrect `[[ma]]`.
`recursive` allow to pick up all values from `noneArrayList`. 
`Print` shows (new line = new start)
```
Array [Bool True,Bool True,Bool True]
Array [Date (Zoned 1974-04-03 22:04:00.000000000051 -0751),Date (Day 2018-04-06),Date (Zoned 2005-04-22 21:14:00.000000000024 -0628)]
and so on
```
So, `genArray` generates any value arrays.